stockroom/external/importer/torchvision_importers.py

```diff
@@ -81,3 +86,325 @@ class FashionMnist(TorchvisionCommon):
+    name = "image_folder"
+
+    def __init__(self, root):
+        root = Path(root)
+        try:
+            self.dataset = datasets.ImageFolder(root)
+        except FileNotFoundError:
+            raise RuntimeError(f"Cannot find the directory '{root.expanduser()}'")
+
+        self.split = root.name
+        self.sample_img, self.sample_label = self._process_data(*self.dataset[0])
+
+        self._gen_metadata()
+        self._print_table()
+
+    def column_names(self):
+        return (
+            f"{self.name}-{self.split}-image",
+            f"{self.name}-{self.split}-label",
+        )
+
+    def shapes(self):
+        return self._shape
+
+    def dtypes(self):
+        return self.sample_img.dtype, self.sample_label.dtype
+
+    @classmethod
+    def gen_splits(cls, root):
+        return [cls(root)]
+
+    def __iter__(self):
+        for img, label in self.dataset:
+            yield self._process_data(img, label)
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def variability_status(self):
+        return True, False
+
+    @staticmethod
+    def _process_data(img, lbl):
+        img = np.array(img)
+        lbl = np.array(lbl)
+        return img, lbl
+
+    def _gen_metadata(self):
+        print("Parsing data...")
+        H = W = 0
+        sample_counter = defaultdict(int)
+        for img, lbl in self.dataset:
+            # find max height/width
+            h, w = img.size
+            if h > H:
+                H = h
+            if w > W:
+                W = w
+
+            # find num samples in each class
+            sample_counter[lbl] += 1
+        _, _, c = self.sample_img.shape
+        self._shape = ((W, H, c), self.sample_label.shape)
+
+    def _print_table(self):
+        table = Table(title="Sample Distribution")
+
+        # columns
+        table.add_column("Class Name")
+        table.add_column("Number of samples", justify="right")
+
+        # rows
+        for class_label in self.dataset.classes:
+            num_samples = self.sample_counter[self.dataset.class_to_idx[class_label]]
+            table.add_row(str(class_label), str(num_samples))
+
+        console = Console()
+        console.print(table)
+
+
+def setup_coco_file(root: Path):
+    """
+    This is a Utility function for the Coco dataset. This function checks for any files
+    downloaded and if not it downloads and sets it up for you.
+    """
+    urls = {
+        "train.zip": "http://images.cocodataset.org/zips/train2017.zip",
+        "val.zip": "http://images.cocodataset.org/zips/val2017.zip",
+        "test.zip": "http://images.cocodataset.org/zips/test2017.zip",
+        "annotations.zip": "http://images.cocodataset.org/annotations/annotations_trainval2017.zip",
+    }
+
+    all_files_present = True
+    for file in urls:
+        if not os.path.exists(root / file):
+            all_files_present = False
+            break
+
+    # if not present download
+    if not all_files_present:
```

Comment:
This will try to download all the files even if the last file is not available. Let's say, for instance, if the download attempt 1 disconnected at the last file, in the second attempt all the files will be downloaded again. 

I'd say, let's check the existence of each file and make a download request there itself instead of keeping a flag like `all_files_present`. What do you think?

Reply:
yes will add that